scanner/fetcher.go

```diff
@@ -108,9 +112,9 @@ func (f *Fetcher) Prepare(ctx context.Context) (*ct.SignedTreeHead, error) {
 	return sth, nil
 }
 
-// Run performs fetching of the Log. Blocks until scanning is complete or
-// context is cancelled. For each successfully fetched batch, runs the fn
-// callback.
+// Run performs fetching of the Log. It blocks until scanning is complete, Stop
+// is invoked, or the passed in context is cancelled. For each successfully
+// fetched batch, runs the fn callback. Must not be called more than once.
```

Comment:
Why is a second method of gracefully stopping the fetcher needed?  Why can't the user just cancel the context?

Reply:
Currently the user does cancel the context, but that cancels all the ongoing fetches. I tough it would be graceful to finish all the started fetches, and then return from Run.